presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -35,12 +36,14 @@
 import static com.facebook.presto.execution.QueryState.QUEUED;
 import static com.facebook.presto.server.QueryStateInfo.createQueryStateInfo;
 import static com.facebook.presto.server.QueryStateInfo.createQueuedQueryStateInfo;
+import static com.facebook.presto.server.security.RoleType.ADMIN;
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 
 @Path("/v1/queryState")
+@RolesAllowed(ADMIN)
```

Comment:
This can also be used in UIs to serve information - should we also open this up to users?